portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,17 +26,29 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  ArrayList<String> list = new ArrayList<>();    
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //Makes array list forn step 3 example
-    ArrayList<String> list = new ArrayList<>();
-    list.add("This is test 1");
-    list.add("This is test 2");
-    list.add("This is test 3");
-    
     response.setContentType("text/html;");
     response.getWriter().println(list);
   }
+  
+  //Uses getParmeter function to obtain user input and inserts that input into 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String text = getParameter(request, "text-input", "");
+    list.add(text);
+    response.sendRedirect("/index.html");
+  }
+  //Obtains user input and returns the value of that input. If nothing is entered a blank space is returned
```

Comment:
Format all method comments as Javadoc, do Javadoc parsers can read it and extract it to automatically generate documentation.

```suggestion
  /**
   * Obtains user input and returns the value of that input.
   * 
   * @returns The value of the input, or defaultValue if name does not exist.
   */
```